### ISSUE TITLE
How to evaluate on new dataset?

### ISSUE THREAD

**Issue by marziehoghbaie** [clks-wzz/FAS-SGTD#9]
Hi,
How can I test on new dataset?
should the files be in video format or should I just give video frames as images?
tnx in advance

**Comment by clks-wzz** [comment]
You should convert video format to frame format for evaluation. Generally speaking, the score of one video is the average of all/sampled frames' scores of the corresponding video.

**Comment by marziehoghbaie** [comment]
TNX, but I have another question, what is the structure of the test data folder? Do I need to put my video frames in a subfolder with a specific name ?

**Comment by clks-wzz** [comment]
> TNX, but I have another question, what is the structure of the test data folder? Do I need to put my video frames in a subfolder with a specific name ?

Sure, it's better to put frames in well designed subfolders with specific names.

**Comment by Roger8** [comment]
> > TNX, but I have another question, what is the structure of the test data folder? Do I need to put my video frames in a subfolder with a specific name ?
> 
> Sure, it's better to put frames in well designed subfolders with specific names.

hello @clks-wzz, great work u have did. About **【well designed subfolders with specific names】** ， Could u please give **an example of the folder structure and file name example** ? Is it like this *test_images/3_1_25_001_scene1.jpg* or sth else ? Looking forward to your reply.

**Comment by 10LGUO** [comment]
> You should convert video format to frame format for evaluation. Generally speaking, the score of one video is the average of all/sampled frames' scores of the corresponding video.

I didn't find any code related to video-frame conversion in your repo. Do I need to write such code on my own?

**Comment by clks-wzz** [comment]
> > You should convert video format to frame format for evaluation. Generally speaking, the score of one video is the average of all/sampled frames' scores of the corresponding video.
> 
> I didn't find any code related to video-frame conversion in your repo. Do I need to write such code on my own?

Yes, you can use python-cv2 or ffmpeg to convert videos to frames.

**Comment by 10LGUO** [comment]
> > > You should convert video format to frame format for evaluation. Generally speaking, the score of one video is the average of all/sampled frames' scores of the corresponding video.
> > 
> > 
> > I didn't find any code related to video-frame conversion in your repo. Do I need to write such code on my own?
> 
> Yes, you can use python-cv2 or ffmpeg to convert videos to frames.

Thanks so much for the reply. May I ask what does the "scene.dat" correspond to in your code and how should I get it from a raw image set.

**Comment by eesaeedkarimi** [comment]
I have the same issue.
1. What should be the names of frame images?
2. How can I pass the address of the directory that includes frame images to the code?
3. Should I compute the depth by PRNet for testing a video?
4. Where should I put the depth output for the test mode? What should be the name of the depth outputs for the test mode?
Is there anybody help me, please?

**Comment by mxbastidasr** [comment]
> I have the same issue.
> 
> 1. What should be the names of frame images?
> 2. How can I pass the address of the directory that includes frame images to the code?
> 3. Should I compute the depth by PRNet for testing a video?
> 4. Where should I put the depth output for the test mode? What should be the name of the depth outputs for the test mode?
>    Is there anybody help me, please?

Hello! Did you figure it out? 
So far, by checking the code, I know that you have a function per dataset on util_dataset.py and that the files names will be managed on the generate_existFaceLists_perfile function on the generate_data_test.py file, but I'm not sure if the depth need to be compute for the testing video with PRNet
Thanks in advance!

**Comment by clks-wzz** [comment]
> > > > You should convert video format to frame format for evaluation. Generally speaking, the score of one video is the average of all/sampled frames' scores of the corresponding video.
> > > 
> > > 
> > > I didn't find any code related to video-frame conversion in your repo. Do I need to write such code on my own?
> > 
> > 
> > Yes, you can use python-cv2 or ffmpeg to convert videos to frames.
> 
> Thanks so much for the reply. May I ask what does the "scene.dat" correspond to in your code and how should I get it from a raw image set.

"scene.dat"  denotes the file which contains the location(bbox) of face. You can make scripts or use face detector to get it.

**Comment by 610821216** [comment]
> > > > > You should convert video format to frame format for evaluation. Generally speaking, the score of one video is the average of all/sampled frames' scores of the corresponding video.
> > > > 
> > > > 
> > > > I didn't find any code related to video-frame conversion in your repo. Do I need to write such code on my own?
> > > 
> > > 
> > > Yes, you can use python-cv2 or ffmpeg to convert videos to frames.
> > 
> > 
> > Thanks so much for the reply. May I ask what does the "scene.dat" correspond to in your code and how should I get it from a raw image set.
> 
> "scene.dat" denotes the file which contains the location(bbox) of face. You can make scripts or use face detector to get it.

May i ask scene.dat is original image  or Depth image???
scene.dat is face position or feature??
and original or Depth input size is 256x256  and 32x32???
if it is possible can give me scene.dat?

**Comment by gabrielvannier** [comment]
Does someone have found how the dataset folder should be named and structured more specifically ? I can not find anything on the path to video frame for testing

**Comment by cuiwenbing** [comment]
use OULU-NPU dataset , What should be the names of frame images?

**Comment by cuiwenbing** [comment]
> Does someone have found how the dataset folder should be named and structured more specifically ? I can not find anything on the path to video frame for testing

use OULU-NPU dataset , What should be the names of frame images?   did you know?

**Comment by cuiwenbing** [comment]
> I have the same issue.
> 
> 1. What should be the names of frame images?
> 2. How can I pass the address of the directory that includes frame images to the code?
> 3. Should I compute the depth by PRNet for testing a video?
> 4. Where should I put the depth output for the test mode? What should be the name of the depth outputs for the test mode?
>    Is there anybody help me, please?

can you answer these questions now?

**Comment by JavanehBahrami** [comment]
> I have the same issue.
> 
> 1. What should be the names of frame images?
> 2. How can I pass the address of the directory that includes frame images to the code?
> 3. Should I compute the depth by PRNet for testing a video?
> 4. Where should I put the depth output for the test mode? What should be the name of the depth outputs for the test mode?
>    Is there anybody help me, please?

@clks-wzz , can you please answer these questions?

**Comment by hollerback370** [comment]
这个项目真的完全不用心，开源是开源，数据集的格式，如何放置，具体推理用的模型完全没说清楚，乱七八糟的

**Comment by KorolLich** [comment]
After a few days of digging through the source code, I think I understand the structure of the OULU-like dataset:
```
FAS-SGDT/train_images/
├── 5_3_20_4/ # the folder name is strictly 4 numbers separated by a "_", contains image sequences from 5_3_20_4.mp4
│   ├── 5_3_20_4_001_scene.jpg # 3 digits exactly, 001 - first frame in video
│   ├── 5_3_20_4_001_scene.dat
│   ├── 5_3_20_4_002_scene.jpg # second frame in video
│   ├── 5_3_20_4_002_scene.dat
│   └── …
├── train_depth/
│   ├── 5_3_20_4/
│   │   ├── 5_3_20_4_001_depth1D.jpg # rgb-image with generated depth from PRNet
│   │   ├── 5_3_20_4_002_depth1D.jpg
│   │   └── …
```

scene.dat contains 4 lines with face bbox coordinates like:
```
x
y
w
h
```

You can check FLAGS.py to edit paths for the flags.path.train_file 

For dev and test dataset structure is the same as for train

However there is a filtering of folders by incorrect numbers according to the type of dataset here: https://github.com/clks-wzz/FAS-SGTD/blob/25b80e52495735537d238fa58d6342c4cd5f9b23/fas_sgtd_single_frame/util/util_dataset.py#L11

For IJCB protocol 1 each folder has template like `{Phone}_{Session}_{User_id}_{PAI}` where
1.  **Train set** must contains `phone` in range [1, 6]; session in [1, 2]; `user_id` in [1, 21); `PAI` in [1, 5]
2.  **Dev set** must contains `phone` in range [1, 6]; session in [1, 2]; `user_id` in [21, 36); `PAI` in [1, 5] 
3.  **Test set** must contains `phone` in range [1, 6]; session in [3]; `user_id` in [36, 56); `PAI` in [1, 5]
